core/src/main/java/io/grpc/internal/AbstractManagedChannelImplBuilder.java

```diff
@@ -308,6 +308,10 @@ public ManagedChannel build() {
       // getResource(), then this shouldn't be a problem unless called on the UI thread.
       nameResolverFactory = NameResolverProvider.asFactory();
     }
```

Comment:
Didn't realize that this was all that AuthorityOverridingTransportFactory was trying to accomplish. Done.